### ISSUE TITLE
publicPath in vue.config works differently from quasar.config

### ISSUE THREAD

**Issue by Neschastje** [quasarframework/quasar#16945]
### What happened?

I tried to build SPA with publicPath: './' to make all assets to have relative path in build as in the [docs](https://cli.vuejs.org/config/#publicpath)
I've got this in my index: 
`
<!DOCTYPE html><html><head><title>Quasar App</title><meta charset=utf-8><meta name=description content="A Quasar Project"><meta name=format-detection content="telephone=no"><meta name=msapplication-tap-highlight content=no><meta name=viewport content="user-scalable=no,initial-scale=1,maximum-scale=1,minimum-scale=1,width=device-width"><link rel=icon type=image/png sizes=128x128 href="/./icons/favicon-128x128.png"><link rel=icon type=image/png sizes=96x96 href="/./icons/favicon-96x96.png"><link rel=icon type=image/png sizes=32x32 href="/./icons/favicon-32x32.png"><link rel=icon type=image/png sizes=16x16 href="/./icons/favicon-16x16.png"><link rel=icon type=image/ico href="/./favicon.ico">  <script type="module" crossorigin src="/./assets/index.c06328dc.js"></script>
  <link rel="stylesheet" href="/./assets/index.89771318.css">
</head><body><div id=q-app></div></body></html>
`

As you can see relative path has /./ prefix.

### What did you expect to happen?

In standard Vue with publicPath: './' you got something like that:
`<!doctype html><html lang=""><head><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width,initial-scale=1"><link rel="icon" href="favicon.ico"><title>vue-static-test</title><script defer="defer" src="js/chunk-vendors.121250b8.js"></script><script defer="defer" src="js/app.204ae534.js"></script><link href="css/app.2cf79ad6.css" rel="stylesheet"></head><body><noscript><strong>We're sorry but vue-static-test doesn't work properly without JavaScript enabled. Please enable it to continue.</strong></noscript><div id="app"></div></body></html>`
There is no such prefix.

### Reproduction URL

https://codepen.io/rstoenescu/pen/xxEvBLR

### How to reproduce?

1. Create new Quasar Project
2. Change publicPath value to './'
3. quasar build
4. Look into dist/index.html

### Flavour

Quasar CLI with Vite (@quasar/cli | @quasar/app-vite)

### Areas

SPA Mode

**Comment by Neschastje** [comment]
Sorry for empty fork in reproduction, I don't know what to write here.

**Comment by rstoenescu** [comment]
A publicPath with relative path is really unusual. It can easily break the app if not using vue router in "hash" mode. Anyway, we'll see what we can do.

**Comment by Neschastje** [comment]
> A publicPath with relative path is really unusual. It can easily break the app if not using vue router in "hash" mode. Anyway, we'll see what we can do.

Yes, my final use case is pretty specific. I'm trying to build SCORM-package from my already existing SPA. Without relative path asset will not be found inside Moodle environment. The only way for me now is manually edit dist.

**Comment by jm-lovelace** [comment]
I ran into this with a particular use case: publishing a Quasar app as a web game on Itch.io. Itch expects relative paths, and there doesn't seem to be a way to get Quasar to use relative paths when building

**Comment by MikeVLyu** [comment]
> A publicPath with relative path is really unusual. It can easily break the app if not using vue router in "hash" mode. Anyway, we'll see what we can do.

Hi, 
The relative path is really needed in case of the application is deployed and managed by an external shell (K8s, docker, etc.) and there is no information about its publicPath while the app is developed. Thus the relative path helps with infrastructure virtualization.

**Comment by rmetcalf9** [comment]
I have a requirement for this issue to be fixed as well

**Comment by rmetcalf9** [comment]
I will provide more details of my problem. 
I am updating an app that ran on an older version of quasar.
I run the quasar dev process and put the /dist files into a container
This container is run in three ways:
1. on my prod server in a path /appname_/v1/asometext/#/*
2. in the ci process during build for testing
3. locally on dev machines in a path /#/*

Number 3 is necessary as the app becomes part of my microservice architecture and devs run it locally while building other parts of the system.

This was all working fine on previous versions of quasar but I am now updating it to a more modern version of quasar:
Here is my quasar version: @quasar/app-webpack 4.3.1 (@quasar/cli 2.5.0)

in the old version of the app the index used to output resourses as follows:
<meta name=msapplication-TileImage content=icons/ms-icon-144x144.png>

Note the path does not start with any slashes (/'s)

In the new version the paths come out as 
<meta name=msapplication-TileImage content=/icons/ms-icon-144x144.png>

Now the paths start with a forward slash (/) so the app is broken as it can not find any of it's resourses.

In experimenting to get this workingI have altered my quasar.config.cjs:
```
    build: {
      vueRouterMode: 'hash', // available values: 'hash', 'history'
      publicPath: '',
```

When publicPath is ''  the url's output like content=/icons/ms-icon-144x144.png
When publicPath is './' the url's output like content=/./icons/ms-icon-144x144.png

Neither work for me.

I have a hacky workaroud. At the start of quasar.config.cjs I put the following:
```
  const overridePublicPath = process.env.OVERRIDE_PUBLIC_PATH
  const publicPathValue = overridePublicPath && overridePublicPath.trim() !== ''
    ? overridePublicPath
    : ''
  console.log('Using publicPath =', publicPathValue)
```
and I set publicPath: publicPathValue

That enables me to build a dev version without specifying any environment and it runs locally
Then build it in my CI process with OVERRIDE_PUBLIC_PATH set to my website url and it works in prod.

The problem is that my container now can no longer work when run locally on the dev machines. I really want it to work with relative paths like it did on the older version of quasar e.g.:
content=icons/ms-icon-144x144.png

with no slashes!

I think that it is well worth fixing this bug with quasar.

In the meantime is there anything I can add to quasar.config.cjs which will cause it to work on relative paths? That will make my container portable again.

**Comment by rstoenescu** [comment]
Fixed in 35a78264f, will be part of the next @quasar/app-vite release.

`build.publicPath: './'` is now passed through to Vite as its relative base, so the build references its assets (and the links in index.html) relative to index.html and can be served from any folder. It requires `build.vueRouterMode: 'hash'` and is available for the SPA and PWA modes only (a relative build cannot work with history mode nor server-side); any other combination is rejected with an explicit error instead of silently producing a root-absolute build. In dev the app is served from the root, same as Vite does.

Public folder files referenced from components should use the `import.meta.env.BASE_URL` prefix, as described in https://v2.quasar.dev/quasar-cli-vite/handling-assets#static-assets-public.
